hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineProvider.java

```diff
@@ -163,6 +174,12 @@ public synchronized Pipeline create(RatisReplicationConfig replicationConfig,
           containerSizeBytes);
       break;
     case THREE:
+      List<DatanodeDetails> excludeDueToEngagement =
+          filterPipelineEngagement();
+      if (excludeDueToEngagement.size() > 0 && excludedNodes.size() == 0) {
+        excludedNodes = excludeDueToEngagement;
+      }
+      excludedNodes.addAll(filterPipelineEngagement());
```

Comment:
Why calling `filterPipelineEngagement()` twice?

Reply:
Thanks.  Mistake corrected.  Code was added to set `excludedNodes` to `excludeDueToEngagement` because the excluded list is immutable when created in the `createPipeline()`.
https://github.com/neils-dev/ozone/blob/76cb3bfed0d3a82f9548c5b233fce6452dd24ffd/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java#L201